js/block.js

```diff
@@ -3674,6 +3674,23 @@ class Block {
                     '" />';
                 labelElem.classList.add("hasKeyboard");
                 this.label = docById("numberLabel");
+                this.label.addEventListener("input", function () {
+                    let input = this.value;
+                    input = parseFloat(input).toString();
+                    console.log("lenght is " + input.length);
+                    if (input % 1 === 0 && input.length >= 6) {
+                        if (input > 0) {
+                            activity.errorMsg(_("Warning: Forward block value should be less than 6 digits long."), blk);
```

Comment:
I think this is a confusing way to express the problem. Checking on the number of digits is how you chose to resolve the issue, but it is not the underlying problem. The underlying problem is the number is to large. So I would rephrase it as "Forward block value should be less than 100000."

Reply:
I cannot think of a use case for numbers > 100000 in MB, so yes, maybe a warning in the number block itself would make more sense. It will still be possible to generate larger numbers through math operators if they are really necessary.

Reply:
@walterbender if the number is a float then even a floats with 5 digits before the decimal hang the browser. Should we only allow 4 digit integers and floats with 4 digits before the decimal or should we have different checks for floats and integers ?

Reply:
@walterbender Despite checking for abs(N) > 100,000, there are instances of lag, particularly with float values having 5 digits before the decimal such as "12345.123232323". These cases pass the abs(N) > 100,000 check but eventually result in browser hang-ups. My suggestion is to revise the threshold to > 10,000, addressing both float-specific and integer-specific concerns for a more optimized solution.

This is how i plan to handle the warning and truncation -
```
else if (newValue >= 100000 || newValue <= -100000) {
                const thisBlock = this.blocks.blockList.indexOf(this);
                if (newValue <= -100000) {
                    activity.errorMsg("Number block value should be more than -100000.", thisBlock);
                    this.value = newValue.slice(0, 6); // truncating 6 places to include the minus sign
                } else {
                    activity.errorMsg("Number block value should be less than 100000.", thisBlock);
                    this.value = newValue.slice(0, 5);
                }
            }
```

Reply:
Is there really a difference in performance (lag) between 12345.1232323 and 12345 ??? If so, we need to understand why. (Maybe 100000 is too much?)

Reply:
@walterbender Upon further inspection of the MB website, I've discovered that entering a number larger than 100,000 in either the Forward or Back block leads to a browser crash. Notably, this issue is confined to these two blocks and does not extend to the entire number block functionality. I apologize for any confusion, and I'll promptly update the pull request to address this specific issue.

Reply:
One other thing to check: does it make a difference whether or not Wrap is enabled? W/O Wrap, the virtual canvas could get to be very large, which could account for much of the performance problem. That said, with Wrap, there will be a lot of wrapping with large numbers, which could also be a problem.

Reply:
@walterbender If I enclose everything within a wrap block with its mode set to off, the website remains stable. However, when the wrap mode is switched to on, it leads to a crash.

Reply:
@walterbender Graphic blocks pose significant challenges with large values, exhibiting highly unpredictable behavior. Initiating a new project and using the forward or back block with larger values causes issues, while other graphic blocks with similar values do not. However, through experimentation and changing the values of other blocks like the pitch blocks, I discovered that almost all graphic blocks hang the website with larger values in certain situations.

Should I just add the warning and truncation function to the number block as a whole? Is there really a need for values >= 100000 in MB?

Reply:
I am still a bit confused. You mentioned earlier a problem with 12345, so maybe 100000 is also too big? We should try to get a sense of where a reasonable limit is. And maybe we should set different limits for wrap and not wrap?

I suspect that the arc and setxy blocks will cause issues too. And perhaps the scroll block??? (Essentially any block with the potential of generating a large virtual canvas.)

But in general, I'd like to let kids explore large numbers and understand their impact. So only cases where it hangs/crashes the program should we apply the restrictions.

Reply:
> I am still a bit confused. You mentioned earlier a problem with 12345, so maybe 100000 is also too big? We should try to get a sense of where a reasonable limit is. And maybe we should set different limits for wrap and not wrap?
The thing is that for larger 5 digit values like 99999, the output on the canvas becomes extremely laggy, but this is not the case for smaller 5 digit values like 12345. So should we just restrict the values to 4 digits?

> I suspect that the arc and setxy blocks will cause issues too. And perhaps the scroll block??? (Essentially any block with the potential of generating a large virtual canvas.)
Yes exactly, graphic blocks are giving problems with large values which I believe is due to the fact that output on the canvas takes longer to draw. Should we  just restrict the number block from having values larger than our restriction?

Reply:
I think we should put the restrictions on the graphics blocks themselves:

forward (and back) < 5000 ???
arc (radius) < 5000 ???
setxy (both x and y) < 5000 ???
scroll (both x and y) < 5000 ???

There are a few places were larger values could still leak in, e.g., the setter functions. But let's not worry about those for right now.

Reply:
sure, i'll get right on it

Reply:
Devin and I were discussing this and Devin had an interesting idea. Not sure how easy this would be to implement, but what if there was no restrictions, but whenever the "pen" leaves some predefined canvas, it doesn't draw, hence it doesn't extend the canvas.

Reply:
@walterbender, Devin's idea sounds intriguing. I'll give it a shot and see if I can come up with the code to implement this concept. I'll keep you posted on my progress.